cmd/acr/purge.go

```diff
@@ -184,20 +197,24 @@ func purgeTags(ctx context.Context, acrClient api.AcrCLIClientInterface, loginUR
 		lastTag = newLastTag
 		skippedTagsCount = newSkippedTagsCount
 		if tagsToDelete != nil {
-			for _, tag := range *tagsToDelete {
-				wg.Add(1)
-				// The purge job is queued, after a purge worker picks it up the tag will be deleted.
-				worker.QueuePurgeTag(loginURL, repoName, *tag.Name, *tag.Digest)
+			for i, tag := range *tagsToDelete {
+				purger.StartPurgeTag(ctx, loginURL, repoName, *tag.Name)
 				deletedTagsCount++
+				// Because the purger ErrChan has a capacity of batchSize it has to periodically be checked for errors
+				if math.Mod(float64(i+1), float64(purger.BatchSize())) == 0 {
+					purger.Wait()
+					purgeErr := purger.FlushErrChan()
+					if purgeErr != nil {
+						return -1, purgeErr
+					}
+				}
 			}
 			// To not overflow the error channel capacity the purgeTags function waits for a whole block of
-			// 100 jobs to be finished before continuing.
-			wg.Wait()
-			for len(worker.ErrorChannel) > 0 {
-				wErr := <-worker.ErrorChannel
-				if wErr.Error != nil {
-					return -1, wErr.Error
-				}
+			// jobs to be finished before continuing.
+			purger.Wait()
+			err = purger.FlushErrChan()
+			if err != nil {
+				return -1, err
 			}
```

Comment:
We probably need to re-design this part. The objective is that we should return error if there is an error.